tests/course/test_for_courses.py

```diff
@@ -0,0 +1,181 @@
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user0.pk)
+        # USER1 AS TEACHER
+        assert studentCourses2.teacher_course_id.course_id in StudentCourses.objects.get_teacher_pending(user1.pk)
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
+        studentCourses2.change_to_confirmed()
+        assert studentCourses1.teacher_course_id.course_id not in StudentCourses.objects.get_teacher_pending(user1.pk)
+
+    def test_student_pending(sself, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS STUDENT
+        assert studentCourses1 not in StudentCourses.objects.get_student_pending(user0)
+        assert studentCourses2 in StudentCourses.objects.get_student_pending(user0)
+        studentCourses2.change_to_confirmed()
+        assert studentCourses2 not in StudentCourses.objects.get_student_pending(user0)
+        # USER1 AS STUDENT
+        assert studentCourses1 in StudentCourses.objects.get_student_pending(user1)
+        assert studentCourses2 not in StudentCourses.objects.get_student_pending(user1)
+        studentCourses1.change_to_confirmed()
+        assert studentCourses1 not in StudentCourses.objects.get_student_pending(user1)
+
+    def test_student_courses(self, studentCourses1, studentCourses2, user0, user1):
+        # USER0 AS STUDENT
+        assert studentCourses1 not in StudentCourses.objects.get_student_courses(user0)
+        assert studentCourses2 in StudentCourses.objects.get_student_courses(user0)
+        # USER1 AS STUDENT
+        assert studentCourses1 in StudentCourses.objects.get_student_courses(user1)
+        assert studentCourses2 not in StudentCourses.objects.get_student_courses(user1)
+
+    # CHECK INVALED DATA INPUT
+
+    def test_regitser(self, db, teacherCourses1, user0, user1):
+        # REGISTER TO YOUR OWN COURSE
+        with pytest.raises(ValidationError):
+            student_course = StudentCourses(student_id=user0, teacher_course_id=teacherCourses1)
+            student_course.save()
+        # REGISTER TO THE SAME COURSE TWICE
+        with pytest.raises(ValidationError):
+            student_course = StudentCourses(student_id=user1, teacher_course_id=teacherCourses1)
+            student_course.save()
+            student_course1 = StudentCourses(student_id=user1, teacher_course_id=teacherCourses1)
+            student_course1.save()
+
+    def test_status(self, db, teacherCourses1, user1):
+        # INVALED STATUS
+        with pytest.raises(ValidationError):
+            student_course = StudentCourses(student_id=user1, teacher_course_id=teacherCourses1,
+                                            status=INVALID_STATUS)
+            student_course.save()
+
+    def test_user(self, db, teacherCourses2, user0):
```

Comment:
a better name will remove the need for the comment, for example: `test_registering_a_student_without_user_fails_validation`